HealthVault/Classes/Caching/Private/MHVThingCacheDatabase.m

```diff
@@ -759,6 +778,231 @@ - (void)updateLastCompletedSyncDate:(NSDate *_Nullable)lastCompletedSyncDate
+        {
+            completion([NSError MHVCacheDeleted]);
+        }
+        return;
+    }
+    
+    [self.managedObjectContext performBlock:^
+    {
+        MHVCachedRecord *record = (MHVCachedRecord *)[self fetchCachedRecord:pendingMethod.recordId.UUIDString];
+        
+        if (!record)
+        {
+            if (completion)
+            {
+                completion([NSError MHVCacheError:@"Record could not be found"]);
+            }
+            
+            return;
+        }
+        
+        // Support for updating a pending method request - If a pending method with the same identifier exists update rather
+        // than creating a new one.
+        MHVPendingThingOperation *operation = [record pendingThingOpertaionWithIdentifier:pendingMethod.identifier];
+        
+        if (!operation)
+        {
+            operation = [self newPendingThingOperationForRecord:record];
+            
+            if (!operation)
+            {
+                if (completion)
+                {
+                    completion([NSError MHVCacheError:@"Could not create a new MHVPendingThingOperation."]);
+                }
+                return;
+            }
+            
+        }
+        
+        operation.name = pendingMethod.name;
+        operation.version = pendingMethod.version;
+        operation.originalRequestDate = pendingMethod.originalRequestDate;
+        operation.parameters = pendingMethod.parameters;
+        operation.identifier = pendingMethod.identifier;
+        operation.correlationId = pendingMethod.correlationId.UUIDString;
+        
+        NSError *error = [self saveContext];
+        
+        if (completion)
+        {
+            completion(error);
+        }
+
+    }];
+}
+
+- (void)fetchPendingMethodsForRecordId:(NSString *)recordId
+                            completion:(void (^)(NSArray<MHVPendingMethod *> *_Nullable methods, NSError *_Nullable error))completion
+{
+    if ([NSString isNilOrEmpty:recordId])
+    {
+        if (completion)
+        {
+            completion(nil, [NSError MVHRequiredParameterIsNil]);
+        }
+        return;
+    }
+    
+    if (!self.isDatabaseReady)
+    {
+        if (completion)
+        {
+            completion(nil, [NSError MHVCacheDeleted]);
+        }
+        return;
+    }
+    
+    [self.managedObjectContext performBlock:^
+    {
+        //Create query to filter & order Things
+        NSPredicate *predicate = [NSPredicate predicateWithFormat:@"record.recordId == %@", [recordId lowercaseString]];
+        
+        NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:@"MHVPendingThingOperation"];
+        fetchRequest.predicate = predicate;
+        fetchRequest.sortDescriptors = @[[NSSortDescriptor sortDescriptorWithKey:@"originalRequestDate" ascending:NO]];
+        
+        NSError *error;
+        NSUInteger fetchCount = [self.managedObjectContext countForFetchRequest:fetchRequest error:&error];
+        if (error)
```

Comment:
It is quite a bit more efficient to just fetch the count if the result is 0. But we would need to do some extended testing to see if it makes sense to do it everywhere.